examples/ex-batched-matrix-multiply.cpp

```diff
@@ -27,264 +27,469 @@
-        Aview(e,row,col) = rand() % 50 + 1; 
-        Bview(e,row,col) = rand() % 50 + 1; 
-        Al2view(e,row,col) = Aview(e,row,col);
-        Bl2view(e,row,col) = Bview(e,row,col);
+  //
+  // Allocate space for data in layout 2
+  //
+  double *Al2 = memoryManager::allocate<double>(NCOLS * NROWS * NMAT);
+  double *Bl2 = memoryManager::allocate<double>(NCOLS * NROWS * NMAT);
+  double *Cl2 = memoryManager::allocate<double>(NCOLS * NROWS * NMAT);
+
+  //
+  // Permuted layout - equivalent to indexing via
+  // A[e + NELEM*(r + NROWS*c)]
+  auto layout2 =
+      RAJA::make_permuted_layout({{NMAT, NROWS, NCOLS}},
+                                 RAJA::as_array<RAJA::Perm<1, 2, 0> >::get());
+  RAJA::View<double, RAJA::Layout<3, Index_type, 0> > Al2view(Al2, layout2);
+  RAJA::View<double, RAJA::Layout<3, Index_type, 0> > Bl2view(Bl2, layout2);
+  RAJA::View<double, RAJA::Layout<3, Index_type, 0> > Cl2view(Cl2, layout2);
+
+  //
+  // Initialize data
+  //
+  for (int e = 0; e < NMAT; ++e) {
+    for (int row = 0; row < NROWS; ++row) {
+      for (int col = 0; col < NCOLS; ++col) {
+        Aview(e, row, col) = rand() % 50 + 1;
+        Bview(e, row, col) = rand() % 50 + 1;
+        Al2view(e, row, col) = Aview(e, row, col);
+        Bl2view(e, row, col) = Bview(e, row, col);
       }
-    }                 
+    }
   }
 
-//-------------------------------------------
-//Matrix multiply with layout 1 on the CPU
-//
-  using CPUPol1 = RAJA::simd_exec;
+  //-------------------------------------------
+  // Matrix multiply with layout 1 on the CPU with loop policy
+  //
+  using CPUPol1 = RAJA::loop_exec;
+
+  myMin = std::numeric_limits<double>::max();
+  for (int i = 0; i < NITER; ++i) {
 
-  myMin = std::numeric_limits<double>::max();  
-  for(Index_type i=0; i<NITER; ++i){
-    
     timer.start();
-    RAJA::forall<CPUPol1>(RAJA::RangeSegment(0, NMAT), [=] (Index_type i) {      
-
-        Cview(i,0,0) = Aview(i,0,0)*Bview(i,0,0) + Aview(i,0,1)*Bview(i,1,0) + Aview(i,0,2)*Bview(i,2,0);
-        Cview(i,0,1) = Aview(i,0,0)*Bview(i,0,1) + Aview(i,0,1)*Bview(i,1,1) + Aview(i,0,2)*Bview(i,2,1);
-        Cview(i,0,2) = Aview(i,0,0)*Bview(i,0,2) + Aview(i,0,1)*Bview(i,1,2) + Aview(i,0,2)*Bview(i,2,2);
-        
-        Cview(i,1,0) = Aview(i,1,0)*Bview(i,0,0) + Aview(i,1,1)*Bview(i,1,0) + Aview(i,1,2)*Bview(i,2,0);
-        Cview(i,1,1) = Aview(i,1,0)*Bview(i,0,1) + Aview(i,1,1)*Bview(i,1,1) + Aview(i,1,2)*Bview(i,2,1);
-        Cview(i,1,2) = Aview(i,1,0)*Bview(i,0,2) + Aview(i,1,1)*Bview(i,1,2) + Aview(i,1,2)*Bview(i,2,2);
-        
-        Cview(i,2,0) = Aview(i,2,0)*Bview(i,0,0) + Aview(i,2,1)*Bview(i,1,0) + Aview(i,2,2)*Bview(i,2,0);
-        Cview(i,2,1) = Aview(i,2,0)*Bview(i,0,1) + Aview(i,2,1)*Bview(i,1,1) + Aview(i,2,2)*Bview(i,2,1);
-        Cview(i,2,2) = Aview(i,2,0)*Bview(i,0,2) + Aview(i,2,1)*Bview(i,1,2) + Aview(i,2,2)*Bview(i,2,2);
-
-      });
+    RAJA::forall<CPUPol1>(RAJA::RangeSegment(0, NMAT), [=](Index_type i) {
```

Comment:
I think it's clearer if you just use the exec policy directly for the template parameter; i.e., no 'using' directive.